prov/gni/src/gnix_av.c

```diff
@@ -85,6 +85,130 @@ static int gnix_verify_av_attr(struct fi_av_attr *attr)
 }
 
 /*
+ * Given a string that contains the string representation of gnix_ep_name.
+ * Parse this string and return the values in a gnix_ep_name structure.
+ */
+int gnix_av_straddr_to_ep_name(char *buf,
+			       struct gnix_ep_name *gnix_ep)
+{
+	char temp_buf[GNIX_AV_MAX_STR_ADDR_LEN];
+	char *int_buf;
+	uint32_t cdm_id;
+	uint32_t cm_nic_cdm_id;
+	uint32_t cookie;
+	uint32_t device_addr;
+	uint32_t name_type;
+	uint32_t version;
+	int ret = FI_SUCCESS;
+
+	if ((buf == NULL) || (gnix_ep == NULL)) {
+		ret = -FI_EINVAL;
+		goto err;
+	}
+
+	strcpy(temp_buf, buf);
+	errno = 0;
+	gnix_ep->gnix_addr.device_addr = 0;
+	gnix_ep->gnix_addr.cdm_id = 0;
+	gnix_ep->name_type = 0;
+	gnix_ep->cm_nic_cdm_id = 0;
+	gnix_ep->cookie = 0;
+
+	/* extract the version component */
+	int_buf = strtok(temp_buf, ":");
```

Comment:
if you wouldn't mind could you use ```strtok_r``` or perhaps ```strsep```?  I'm thinking of thread safety.